src/arrayMethodJoin.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== 0) {
+        result += separator === undefined ? ',' : separator;
```

Comment:
you should probably keep both check in if statement but that also works 😄 
```suggestion
      if (i !== 0 && separator !== undefined) {
        result += separator
```